addons/mass_mailing_sms/models/sms_tracker.py

```diff
@@ -47,7 +47,7 @@ def _update_sms_traces(self, trace_status, failure_type=False, failure_reason=Fa
                 'trace_status': trace_status,
                 'failure_type': failure_type,
                 'failure_reason': failure_reason,
-                **{'sent_datetime': fields.Datetime.now() if trace_status == 'pending' else {}},
+                'sent_datetime': fields.Datetime.now() if trace_status in ['pending', 'sent'] else {},
```

Comment:
Would guess the intent was probably `**({'sent_datetime': fields.Datetime.now()} if trace_status else {})` so that it wouldn't try to update the value, instead of trying to insert `{}` which should probably fail

Also the sent_datetime of a sent datetime shouldn't be updated to "now", right? Shouldn't it be set somewhere else?

Reply:
As I said on the task, my opinion is:

> Technically, I would suggest updating `_update_sms_traces` to **split the write in two**:
> * for traces currently without `sent_datetime` we'd systematically set it (unless the new status is `process` of course)
> * one for the others where `sent_datetime` is already set (we shouldn't change it).

> [...] the same probably applies to notifications.

@tde-banana-odoo, indeed it should probably be a fix.

Reply:
globally agree :)